speechbrain/core.py

```diff
@@ -822,6 +822,13 @@ def __init__(  # noqa: C901
                         "all other subprocess were killed."
                     )
 
+        if self.device == "cpu" and self.distributed_backend == "nccl":
+            sb.utils.distributed.DistributedState(self.device)
+        else:
+            sb.utils.distributed.DistributedState(
+                self.device, self.distributed_backend
+            )
```

Comment:
I removed the class DistributedState since I agree that torch.distributed is already providing the needed features. At the beginning I was thinking about using this class for the future if we want to support other backend such as TPU, Megatron etc but I think it's not worth it to do it now. (the idea was to abstract some key functions that can differ depending on the backend used).